### PR TITLE
qsv: 0.131.1 -> 0.138.0, build simplification

### PR DESCRIPTION
**Pull request by itepastra**
Update qsv to version 0.138.0, also fix build

ZHF: #352882 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by uncenter** [comment]
I'm getting this test error on darwin:

failures:

---- test_clipboard::clipboard_success stdout ----
thread 'test_clipboard::clipboard_success' panicked at tests/test_clipboard.rs:22:42:
called `Result::unwrap()` on an `Err` value: ClipboardNotSupported - "The selected clipboard is not supported with the current system configuration."
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

File: pkgs/by-name/qs/qsv/package.nix [hunk 1]
```diff
@@ -21,17 +21,21 @@ rustPlatform.buildRustPackage {
...
       "dynfmt-0.1.5" = "sha256-/SrNOOQJW3XFZOPL/mzdOBppVCaJNNyGBuJumQGx6sA=";
-      "grex-1.4.5" = "sha256-4Tr5L87HuiUW8tJdqr4oT1yQXviU7XtDrKY7iYNcwbo=";
+      "grex-1.4.5" = "sha256-g8yPcnDV7hu/K8Ib8j8n+Qz9O7G4zdbMnQopBBn4mr0=";
       "local-encoding-0.2.0" = "sha256-ThXYKr3u/n2kvINcyobB2Ayex2sNbJEOyyjZH993Z4U=";
-      "polars-0.41.3" = "sha256-8xkcJgfKo7BzeBnR6XVdbcH9ZY9Kh4dcGFMmyA5LuQg=";
+      "polars-0.44.2" = "sha256-MqQoHLluWWkWFOK4+Njq/NnYm30XGze9S9cJ7ysPLOw=";
+      "publicsuffix-2.2.3" = "sha256-CV7ey5v1Lma+7q7dL4UfwSy7GjtK/OZvmrbyexFOfBY=";
+      "strum-0.26.3" = "sha256-R3bPqTwxyts7uYSas/3Ij39/m64DH0A++DMIdYqmSqA=";
     };
   };
```
Comment (uncenter) [inline pkgs/by-name/qs/qsv/package.nix:40]:
```nix
  useFetchCargoVendor = true;
  cargoHash = "sha256-tu9HCFAxmmYVgmJyHunBtGSqKGzwbX2vi6ju4cv33wc=";
```

Using this for me locally worked like a charm. Enabled by https://github.com/NixOS/nixpkgs/pull/349360.

Comment (itepastra) [reply pkgs/by-name/qs/qsv/package.nix:40]: I forgot it existed, will change to use the library function

File: pkgs/by-name/qs/qsv/package.nix [hunk 2]
```diff
@@ -21,19 +21,11 @@ rustPlatform.buildRustPackage {
...
-      "dynfmt-0.1.5" = "sha256-/SrNOOQJW3XFZOPL/mzdOBppVCaJNNyGBuJumQGx6sA=";
-      "grex-1.4.5" = "sha256-4Tr5L87HuiUW8tJdqr4oT1yQXviU7XtDrKY7iYNcwbo=";
-      "local-encoding-0.2.0" = "sha256-ThXYKr3u/n2kvINcyobB2Ayex2sNbJEOyyjZH993Z4U=";
-      "polars-0.41.3" = "sha256-8xkcJgfKo7BzeBnR6XVdbcH9ZY9Kh4dcGFMmyA5LuQg=";
-    };
-  };
+  useFetchCargoVendor = true;
+  cargoHash = "sha256-tu9HCFAxmmYVgmJyHunBtGSqKGzwbX2vi6ju4cv33wc=";
 
   buildInputs =
```
Comment (uncenter) [inline pkgs/by-name/qs/qsv/package.nix:30]:
```suggestion
  buildInputs = [
    file
    sqlite
    zstd
  ];
```

Sorry for another non-version related change, just good to simplify this package I think. See https://github.com/NixOS/nixpkgs/issues/354146, we no longer need to specify darwin framework to include!

**Comment by itepastra** [comment]
Like this? @uncenter

**Comment by uncenter** [comment]
Yep that looks exactly right! Thank you!

**Comment by uncenter** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 356504`

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>qsv</li>
  </ul>
</details>

**Review by uncenter — APPROVED** [review]

**Comment by uncenter** [comment]
> It seems to use [arboard](https://github.com/1Password/arboard) for it's clipboard, which is made for X11 and thus won't work on darwin. I'll add it as a disabled test on darwin

Interestingly the clipboard feature seems to work, but the test doesn't. Odd!

**Comment by itepastra** [comment]
In the test it explicitly tries to call a library that is made fox X11, maybe it's real code doesn't do this. Am uncertain

**Comment by uncenter** [comment]
No worries yeah.

**Comment by uncenter** [comment]
Wrong button oops 😓

**Review by detroyejr — APPROVED** [review]
This looks good to me and works on nixos! I have a slight preference to sort the skipped tests within each comment section to keep those grouped, but it's not super important.

@uncenter thanks for the cargo lock file suggestion. Wasn't aware of that and I'll be using that in the future!

**Comment by uncenter** [comment]
Can I get these GitHub actions checks re-run? Looks like they all failed when I accidentally closed this, but they should be passing fine.

**Comment by itepastra** [comment]
I just rebased the last commit to rerun the tests, didn't change anything

**Review by uncenter — APPROVED** [review]

**Review by detroyejr — APPROVED** [review]

**Pull request merged**